### PR TITLE
Update display_list_image_filter_unittests to be permit Skia roll

### PR DESCRIPTION
**Pull request by lhkbob**
https://skia-review.googlesource.com/c/skia/+/571798 had to be reverted because it causes DisplayListImageFilter.LocalImageFilterBounds to fail. With SkMatrixTransformImageFilter (returned by SkImageFilters::MatrixTransform) supporting complex matrices, makeWithLocalMatrix(rotation) will no longer return null.

This is at odds with the behavior of DlImageFilter's implementation which tries to mimic the existing behavior of image filters. This relaxes the test case so that Skia can improve its implementation without breaking Flutter's tests. Flutter can then update DlImageFilter to match out-of-band.

https://bugs.chromium.org/p/skia/issues/detail?id=9283
https://github.com/flutter/flutter/issues/114723

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: display_list/display_list_image_filter_unittests.cc
Comment (flar) [inline display_list/display_list_image_filter_unittests.cc:814]: // Temporarily relax the equivalence testing to allow Skia to expand their
// behavior.
// Once the Skia fixes are rolled in, the DlImageFilter should adapt to the
// new rules. See https://github.com/flutter/flutter/issues/114723

Comment (lhkbob) [reply display_list/display_list_image_filter_unittests.cc]: Done, thanks for creating the flutter issue. I've added it to the PR description as well.

**Review by flar — APPROVED** [review]
LGTM

Thanks for the comment update!

**Comment by flar** [comment]
The rebase has passed all Windows test targets. so I'm adding the autosubmit label

**Review by zanderso — APPROVED** [review]

**Comment by zanderso** [comment]
This needed a second review to make the autosubmit bot happy. I approved, but I'll let @flar add back the label if/when appropriate.

**Pull request merged**
